src/coreclr/jit/morph.cpp

```diff
@@ -9180,6 +9171,13 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optA
 
         tree->AsOp()->gtOp1 = op1 = fgMorphTree(op1, mac);
 
+        if ((mac != nullptr) && mac->m_used && tree->OperIsIndir())
+        {
+            // This context was used to elide an explicit null check on a FIELD_ADDR, with the expectation that
+            // this indirection will now be responsible for null-checking (implicitly).
+            tree->gtFlags &= ~GTF_IND_NONFAULTING;
```

Comment:
Where are we setting these to non-faulting in the first place? Isn't that odd?

Reply:
`FIELD_ADDR` by itself cannot be null, since it null-checks the source object, so it is correct to consider the indirection itself non-faulting at the time of its creation. It's just the special mechanism of morph's that changes its semantics.